scala-package/core/src/main/scala/ml/dmlc/mxnet/Optimizer.scala

```diff
@@ -71,6 +119,10 @@ abstract class Optimizer extends Serializable {
   // Dispose the state it created
   def disposeState(state: AnyRef): Unit
 
+  def serializeState(state: AnyRef): Array[Byte]
+
+  def deserializeState(bytes: Array[Byte]): AnyRef
```

Comment:
I see that when the user wants to implement these two APIs, they have to explicitly convert to/from AnyRef, is it a better choice to make it generic?